graylog2-server/src/main/java/org/graylog2/users/UserImpl.java

```diff
@@ -172,8 +173,9 @@ public void setEmail(final String email) {
 
     @Override
     public void setPermissions(final List<String> permissions) {
+        final List<String> perms = Lists.newArrayList(permissions);
         // Do not store the dynamic user self edit permissions
-        permissions.removeAll(this.permissions.userSelfEditPermissions(getName()));
+        perms.removeAll(this.permissions.userSelfEditPermissions(getName()));
```

Comment:
![I can has unit test for this?](https://cloud.githubusercontent.com/assets/43951/17108018/fa285368-5291-11e6-8b10-1627b177f6c3.jpg)